src/js/player.js

```diff
@@ -1747,10 +1747,11 @@ class Player extends Component {
    */
   duration(seconds) {
     if (seconds === undefined) {
-      return this.cache_.duration || 0;
+      // return NaN if the duration is not known
+      return this.cache_.duration !== undefined ? this.cache_.duration : NaN;
     }
 
-    seconds = parseFloat(seconds) || 0;
+    seconds = parseFloat(seconds);
```

Comment:
I think the problem with leaving the `|| 0` is that if `player.techGet_('duration')` equals `NaN` when [handleTechDurationChange_()](https://github.com/videojs/video.js/blob/cc6e82442fcefb3b62a6e33a588d7f0b7152fd3d/src/js/player.js#L1368) is first called, then the initial value of `player.cache_.duration` is 0 rather than `NaN`. At least that was the case with the page I was testing with.